rules/java/lang/insecure_allow_origin.yml

```diff
@@ -0,0 +1,59 @@
+imports:
+  - java_shared_lang_instance
+  - java_shared_lang_servlet_request
+  - java_shared_lang_user_input
+patterns:
+  - pattern: |
+      $<RES>.$<METHOD>($<HEADER>, $<USER_INPUT>);
+    filters:
+      - variable: RES
+        detection: java_shared_lang_instance
+        scope: cursor
+        filters:
+          - variable: JAVA_SHARED_LANG_INSTANCE_TYPE
+            regex: \A(javax\.servlet\.http\.)?HttpServletResponse\z
+      - variable: METHOD
+        values:
+          - setHeader
+          - addHeader
+      - variable: HEADER
+        string_regex: \A(?i)(Access-Control-Allow-Origin)\z
+      - variable: USER_INPUT
+        detection: java_lang_insecure_allow_origin_user_input
+auxiliary:
+  - id: java_lang_insecure_allow_origin_user_input
+    patterns:
+      - pattern: $<USER_INPUT>;
+        filters:
+          - variable: USER_INPUT
+            detection: java_shared_lang_user_input
+            scope: cursor
+      - pattern: $<USER_INPUT_REQUEST>.$<REQUEST_GET_METHOD>().getAttribute();
+        filters:
+          - variable: USER_INPUT_REQUEST
+            detection: java_shared_lang_servlet_request
+            scope: cursor
+          - variable: REQUEST_GET_METHOD
+            values:
+              - getSession
+              - getServletContext
```

Comment:
@cfabianski or @didroe do you think this is general enough to be added to the shared user input rule?

Reply:
I would say yes!

Reply:
Thanks @cfabianski - I've simplified it a bit too and just added the two methods (`getSession`, `getServletContext`) to the existing list: https://github.com/Bearer/bearer-rules/pull/220/commits/b06bcda0dc1201251bc54e9c120514c9c802a9ac

Reply:
Will merge for now but @didroe if you feel otherwise, let me know and we can revert 👍